src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,106 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (todoId: number) => void,
+  changeCheckbox: (id: number, value: boolean) => void,
+  changeTitle: (id: number, newTitle: string) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  changeCheckbox,
+  changeTitle,
+}) => {
+  const { id, title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [titleEdit, setTitleEdit] = useState(title);
+
+  const onDeleteTodo = () => {
+    deleteTodo(id);
+  };
+
+  const onChangeCheckbox = () => {
+    changeCheckbox(id, completed);
+  };
+
+  const onEditTitle = () => {
+    setIsEditing(true);
+  };
+
+  const handleChangeTitleBlur = () => {
+    if (titleEdit === '') {
+      deleteTodo(id);
+
+      return;
+    }
+
+    if (title !== titleEdit) {
+      changeTitle(id, titleEdit);
+    }
+
+    setIsEditing(false);
+  };
+
+  const onChangeTitleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleEdit(event.target.value);
+  };
+
+  const handleKeyUp = (e: React.KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      setIsEditing(false);
+      setTitleEdit(title);
+    }
+
+    if (e.key === 'Enter' || e.key === '') {
+      handleChangeTitleBlur();
+    }
+  };
+
+  return (
+    <li
+      key={todo.id}
```

Comment:
The `key` attribute is not required for individual component elements and should be removed.

Reply:
updated